TypeSystem/flow-analysis/reachability_for_A01_t01.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion for statement: If `N` is a for statement of the form
+/// `for (D; C; U) S,` then:
+/// - Let `before(D) = before(N)`.
+/// - Let `before(C) = conservativeJoin(after(D), assignedIn(N), capturedIn(N))`
+/// - Let `before(S) = split(true(C))`.
+/// - Let `before(U) = merge(after(S), continue(S))`.
+/// - Let `after(N) = inheritTested(join(false(C), unsplit(break(S))), after(U))`
+///
+/// @description Checks that
+/// `before(C) = conservativeJoin(after(D), assignedIn(N), capturedIn(N))`. Test
+/// that if the static type of `D` is `Never` then `C` is dead code and an
+/// assignment in it will never happen.
+/// @author sgrekhov22@gmail.com
+/// @issue 60089
+
+Never foo() => throw "Never";
+
+test1() {
+  late int i;
+  for (foo(); (i = 42) < 0;) {
+  }
+  i; // Definitely unassigned.
+//^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+test2() {
+  late int i;
```

Comment:
The same kind of wrapping should work here, and in the corresponding location below.